sdk/core/azure-core/src/test/java/com/azure/android/core/util/Base64UrlTest.java

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.util;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static com.azure.android.core.util.Base64Stub.DECODED_BYTES;
+import static com.azure.android.core.util.Base64Stub.ENCODED_BYTES;
+import static com.azure.android.core.util.Base64Stub.ENCODED_STRING;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class Base64UrlTest {
+    @BeforeClass
+    public static void setUp() {
+        Base64Util.setBase64Wrapper(new Base64Stub());
```

Comment:
Is a wrapper being set as Base64 isn't consistent between Android versions or between devices?

Reply:
The reason to set this here is that we cannot call Android framework's classes like Base64 from unit tests and we cannot use Java's Base64 class because that's only available at a later Android version. To get around this and not use something like PowerMock to mock static methods like Base64.encode(), I created the [Base64Wrapper](https://github.com/Azure/azure-sdk-for-android/blob/67845f0155a5700833dae754fbd7c710ca8aaddb/sdk/core/azure-core/src/main/java/com/azure/android/core/util/Base64Util.java#L94) interface. Normally it will always be set to [Base64Android](https://github.com/Azure/azure-sdk-for-android/blob/67845f0155a5700833dae754fbd7c710ca8aaddb/sdk/core/azure-core/src/main/java/com/azure/android/core/util/Base64Util.java#L101) and for unit tests it will be set to [Base64Stub](https://github.com/Azure/azure-sdk-for-android/blob/67845f0155a5700833dae754fbd7c710ca8aaddb/sdk/core/azure-core/src/test/java/com/azure/android/core/util/Base64Stub.java).